app/controllers/api/v1/notes_controller.rb

```diff
@@ -0,0 +1,47 @@
+module Api
+  module V1
+    class NotesController < ApplicationController
+      def index
+        return render_invalid_note_type if invalid_note_type?
+        render json: notes_filtered, status: :ok, each_serializer: IndexNoteSerializer
+      end
+
+      def show
+        render json: show_note, status: :ok, serializer: ShowNoteSerializer
+      end
+
+      private
+
+      def notes
+        Note.all
+      end
+
+      def invalid_note_type?
+        note_type_present? && note_type_ok?
+      end
+
+      def note_type_present?
+        params[:note_type].present?
+      end
+
+      def note_type_ok?
+        !Note.note_types.keys.include?(params[:note_type])
+      end
```

Comment:
Acá como regla general solemos hacer métodos "positivos", asique en lugar de preguntar si es incorrecto dentro del método, preguntaría si es válido:
```suggestion
      def note_type_ok?
        Note.note_types.keys.include?(params[:note_type])
      end
```
Luego arriba en el método `invalid_note_type?` la lógica interna te quedaría tal cual, pero esto lo que te respondería es si es tipo es válido (tendríamos que ajustar el nombre para que refleje la nueva lógica). Asique en el chequeo dentro del servicio podemos cambiar el `if` por un `unless`, que leyéndolo nos queda algo asi como "Devolveme error salvo que sea correcto el tipo de nota". Este uso del `unless` lo vas a ver muchisimo, aunque cuesta agarrarle la mano al principio :cowboy_hat_face:

Reply:
Ahi lo cambie para que funcione positivamente